File: src/main/java/com/munzenberger/feed/log/Logger.java
package com.munzenberger.feed.log;

public interface Logger {

	void addAppender(Appender appender);

	void log(String message);

	void log(String tag, String message);

	void log(String message, Throwable t);

	void log(String tag, String message, Throwable t);
}

File: core/src/main/kotlin/com/munzenberger/feed/Logger.kt
package com.munzenberger.feed

interface Logger {
    fun print(obj: Any)

    fun println(obj: Any)

    fun printStackTrace(t: Throwable)

    fun format(
        format: String,
        vararg params: Any,
    ) {
        val str = String.format(format, *params)
        print(str)
    }

    fun formatln(
        format: String,
        vararg params: Any,
    ) {
        val str = String.format(format, *params)
        println(str)
    }
}
